src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OIDCLoginBackingBean.java

```diff
@@ -0,0 +1,169 @@
+                if (!systemConfig.isSignupDisabledForRemoteAuthProvider(provider.getId())) {
+                    final JwtClaims claims = openIdContext.getAccessToken().getJwtClaims();
+                    final String firstName = claims.getStringClaim(OpenIdConstant.GIVEN_NAME).orElse("");
+                    final String lastName = claims.getStringClaim(OpenIdConstant.FAMILY_NAME).orElse("");
+                    final String verifiedEmailAddress = getVerifiedEmail();
+                    final String emailAddress = verifiedEmailAddress == null ? "" : verifiedEmailAddress;
+                    final String affiliation = claims.getStringClaim("affiliation").orElse("");
+                    final String position = claims.getStringClaim("position").orElse("");
+                    final OAuth2UserRecord userRecord = new OAuth2UserRecord(
+                            provider.getId(),
+                            subject,
+                            claims.getStringClaim(OpenIdConstant.PREFERRED_USERNAME).orElse(subject),
+                            null,
+                            new AuthenticatedUserDisplayInfo(firstName, lastName, emailAddress, affiliation, position),
+                            List.of(emailAddress));
+                    logger.log(Level.INFO, "redirect to first login: " + userRecordIdentifier);
+                    newAccountPage.setNewUser(userRecord);
+                    Faces.redirect("/oauth2/firstLogin.xhtml");
+                }
+            } else {
+                dvUser = userService.updateLastLogin(dvUser);
+                session.setUser(dvUser);
+                Faces.redirect("/");
+            }
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Setting user failed: " + e.getMessage());
+        }
+    }
+
+    public UserRecordIdentifier getUserRecordIdentifier() {
+        try {
+            final String subject = openIdContext.getSubject();
+            final OIDCAuthProvider provider = getProvider();
+            return new UserRecordIdentifier(provider.getId(), subject);
+        } catch (final Exception ignore) {
+            return null;
+        }
+    }
+
+    private String getVerifiedEmail() {
+        try {
+            if (openIdContext.getAccessToken().isExpired()) {
+                return null;
+            }
+            final Object emailVerifiedObject = openIdContext.getClaimsJson().get(OpenIdConstant.EMAIL_VERIFIED);
+            final boolean emailVerified;
+            if (emailVerifiedObject instanceof JsonValue) {
+                final JsonValue v = (JsonValue) emailVerifiedObject;
+                emailVerified = JsonValue.TRUE.equals(v)
+                        || (JsonValue.ValueType.STRING.equals(v.getValueType())
+                                && Boolean.getBoolean(((JsonString) v).getString()));
+            } else {
+                emailVerified = false;
+            }
+            if (!emailVerified) {
+                logger.log(Level.SEVERE,
```

Comment:
Why would that be a _severe_ error message?

Reply:
I would assume that the tenant should not allow unverified emails? This would mean totally anonymous access, you could just log in as anybody and spoof their identity. I thought it might be important. I changed it to FINE now, as it is not really an error.